### DISCUSSION TITLE
Duplicated Commit Message Prefixes

### DISCUSSION THREAD

**Discussion by Shamshiel** [renovatebot/renovate#16260] [Request Help] [answered]
Since yesterday all PRs in my Azure DevOps Repo that were created or updated by my Renovate Bot have a duplicated `commitMessagePrefix`.

As an example: 

Before: 
> PATCH: Update dependency typescript to v4.7.4

After:
> PATCH: pATCH: Update dependency typescript to v4.7.4

All my PRs look like the example above now.

My config looks like this:
```
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": ["config:base", ":semanticCommitScopeDisabled", "docker:enableMajor"],
    "commitMessagePrefix": "PATCH:",
    "semanticCommitScope": null,
    "semanticCommits": "enabled",
    "automerge": false,
    "major": {
        "automerge": false,
        "commitMessagePrefix": "PATCH:"
    },
    "baseBranches": ["develop"]
}
```

Did something change on how commit messages are prefixed? Do I have to change my config to fix my PR titles? 

Maybe it has something to do with the release https://github.com/renovatebot/renovate/compare/32.99.2...32.99.3 because there were some changes on how commit messages are handled?

**Comment by viceice** [comment, marked as answer]
another regression 😞 
- #16239

@pret-a-porter Can you fix it fast? Or do we need to revert again?

**Reply by rarkins** [reply]
I'm reverting: #16263

**Reply by rarkins** [reply]
@pret-a-porter we can't break again. maybe time to throw away this complicated class based logic if it's so easy to get wrong

**Reply by pret-a-porter** [reply]
Sorry, I was out for a really long period of time. The class itself covered by unit tests very well, problem in `generate.ts` script. There is many cases not covered by unit tests. I would try again, if you don't mind.

**Reply by viceice** [reply]
sure. but first add tests to cover existing behavior in a seperate PR.

**Reply by pret-a-porter** [reply]
Make sense. I will do
